From 08912f816f77c5e01d2590c036af1685e994f4e2 Mon Sep 17 00:00:00 2001
From: Vita Preskovsky <vitap@ti.com>
Date: Thu, 7 Jun 2012 19:17:54 +0530
Subject: [PATCH 1025/1092] ARM: OMAP2+: am335x: WLAN and Bluetooth support
 for EVM-SK

Configure SDIO card to support WLAN and UART1 to support
Bluetooth on AM335x EVM-SK.

Signed-off-by: Vita Preskovsky <vitap@ti.com>
[vaibhav.bedia@ti.com - Fix a few checkpatch errors, update
the $SUBJECT to be in line with the general convention and
make use of the evm_id variable for board specific init]
Signed-off-by: Vaibhav Bedia <vaibhav.bedia@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 27 ++++++++++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 6423b7d..0bcadd7 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -905,6 +905,7 @@ static struct pinmux_config ecap2_pin_mux[] = {
 
 #define AM335XEVM_WLAN_PMENA_GPIO	GPIO_TO_PIN(1, 30)
 #define AM335XEVM_WLAN_IRQ_GPIO		GPIO_TO_PIN(3, 17)
+#define AM335XEVM_SK_WLAN_IRQ_GPIO      GPIO_TO_PIN(1, 29)
 
 struct wl12xx_platform_data am335xevm_wlan_data = {
 	.irq = OMAP_GPIO_IRQ(AM335XEVM_WLAN_IRQ_GPIO),
@@ -939,6 +940,13 @@ static struct pinmux_config wl12xx_pin_mux[] = {
 	{NULL, 0},
  };
 
+static struct pinmux_config wl12xx_pin_mux_sk[] = {
+	{"gpmc_wpn.gpio0_31", OMAP_MUX_MODE7 | AM33XX_PIN_OUTPUT},
+	{"gpmc_csn0.gpio1_29", OMAP_MUX_MODE7 | AM33XX_PIN_INPUT},
+	{"mcasp0_ahclkx.gpio3_21", OMAP_MUX_MODE7 | AM33XX_PIN_OUTPUT},
+	{NULL, 0},
+};
+
 static bool backlight_enable;
 
 static void enable_ecap0(int evm_id, int profile)
@@ -1605,6 +1613,13 @@ static void mmc1_init(int evm_id, int profile)
 static void mmc1_wl12xx_init(int evm_id, int profile)
 {
 	setup_pin_mux(mmc1_common_pin_mux);
+	am335x_mmc[1].mmc = 2;
+	am335x_mmc[1].name = "wl1271";
+	am335x_mmc[1].caps = MMC_CAP_4_BIT_DATA | MMC_CAP_POWER_OFF_CARD;
+	am335x_mmc[1].nonremovable = true;
+	am335x_mmc[1].gpio_cd = -EINVAL;
+	am335x_mmc[1].gpio_wp = -EINVAL;
+	am335x_mmc[1].ocr_mask = MMC_VDD_32_33 | MMC_VDD_33_34; /* 3V3 */
 }
 
 static void mmc2_wl12xx_init(int evm_id, int profile)
@@ -1657,6 +1672,15 @@ static void wl12xx_init(int evm_id, int profile)
 	struct omap_mmc_platform_data *pdata;
 	int ret;
 
+	if (evm_id == EVM_SK) {
+		am335xevm_wlan_data.wlan_enable_gpio = GPIO_TO_PIN(0, 31);
+		am335xevm_wlan_data.bt_enable_gpio = GPIO_TO_PIN(3, 21);
+		am335xevm_wlan_data.irq =
+				OMAP_GPIO_IRQ(AM335XEVM_SK_WLAN_IRQ_GPIO);
+		setup_pin_mux(wl12xx_pin_mux_sk);
+	} else {
+		setup_pin_mux(wl12xx_pin_mux);
+	}
 	wl12xx_bluetooth_enable();
 
 	if (wl12xx_set_platform_data(&am335xevm_wlan_data))
@@ -1681,7 +1705,6 @@ static void wl12xx_init(int evm_id, int profile)
 		goto out;
 	}
 
-	setup_pin_mux(wl12xx_pin_mux);
 
 	pdata->slots[0].set_power = wl12xx_set_power;
 out:
@@ -2003,6 +2026,8 @@ static struct evm_dev_cfg evm_sk_dev_cfg[] = {
 	{gpio_led_init,  DEV_ON_BASEBOARD, PROFILE_ALL},
 	{lis331dlh_init, DEV_ON_BASEBOARD, PROFILE_ALL},
 	{mcasp1_init,   DEV_ON_BASEBOARD, PROFILE_ALL},
+	{uart1_wl12xx_init, DEV_ON_BASEBOARD, PROFILE_ALL},
+	{wl12xx_init,       DEV_ON_BASEBOARD, PROFILE_ALL},
 	{NULL, 0, 0},
 };
 
-- 
1.7.11.2

